eclipse-sirius/sirius-components
# Prevent edges from blinking while moving a node or while the diagram refresh

### Type of this PR 


### Issue(s)

closes https://github.com/eclipse-sirius/sirius-components/issues/1130
closes #966
closes #1065
closes #1148 

### What does this PR do?

This PR overrides the default Sprotty edge router and the associated anchor computer to prevent the edge from blinking while the diagram is refreshed.
It handles straight edges between two elements and self-loop edges.
It also improves self-loop edges' position while moving the source/target element.

### Potential side effects

It reuses the backend edge layout algorithm. That means if the backend changes the way edges are layout, those changes will probably need to be reported in the frontend to prevent edges from blinking again.

### How to test this PR?

  - In a domain diagram, create some entities and create some edges.
  - Verify that the diagram refresh does not make edges blink.
  - Verify that moving an element used as a source/target of a self-loop edge does not make that self-loop edge blinking.
